src/engine/TransitivePathImpl.h

```diff
@@ -290,8 +290,9 @@ class TransitivePathImpl : public TransitivePathBase {
     // Make sure we retrieve the Id from an IndexScan, so we don't have to pass
     // this LocalVocab around. If it's not present then no result needs to be
     // returned anyways.
-    if (const Id* id = edges.getEquivalentId(startId)) {
-      result.insert(*id);
+    const auto& idsWithGraph = edges.getEquivalentIdAndMatchingGraphs(startId);
+    for (const auto& [id, graphId] : idsWithGraph) {
```

Comment:
Is for now an invariant that the size of the equivalent bla is `<= 1` (you have a similar check for other places you have to change when the graph support is added).